### PR TITLE
[BP-23] Add options for ETC support

### PR DESCRIPTION
**Pull request by sgregglives**
Signed-off-by: Steven Gregg <steven.gregg@consensys.net>

## PR description

- Add new network options for Ethereum Classic
- Alphabetize options list

## Fixed Issue(s)

BP-23

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -659,25 +689,23 @@ network="rinkeby"
...
-`dev`
-:   PoW development network with a very low difficulty to enable local CPU mining.
-
-!!!note
-    Values are case insensitive, so either `mainnet` or `MAINNET` works.
+!!!tip
+    Values are case insensitive, so either `mainnet` or `MAINNET` works. 
+    
+!!!important
+    The [`--network`](#network) and [`--genesis-file`](#genesis-file) option cannot be used at the same time.
```
Comment (MadelineMurray) [inline docs/Reference/CLI/CLI-Syntax.md]: This tip is duplicated - one needs removing and an _option_ changed to _options_ in the tip that's retained

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -1108,99 +1138,100 @@ Enables [pruning](../../Concepts/Pruning.md) to reduce storage required for the
 !!! important
     Using pruning with [private transactions](../../Concepts/Privacy/Privacy-Overview.md) is not supported.
```
Comment (MadelineMurray) [inline docs/Reference/CLI/CLI-Syntax.md]: The next three options after `--pruning-enabled` are indented inside the important box. If you publish locally, you'll be able to see what I mean.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 3]
```diff
@@ -1108,99 +1138,100 @@ Enables [pruning](../../Concepts/Pruning.md) to reduce storage required for the
...
 
-```bash tab="Syntax"
---remote-connections-max-percentage=<DOUBLE>
-```
-
-```bash tab="Command Line"
---remote-connections-max-percentage=25
-```
+    ```bash tab="Syntax"
+    --required-block=<BLOCK=HASH>
```
Comment (MadelineMurray) [inline docs/Reference/CLI/CLI-Syntax.md]: I know this change wasn't part of this ticket but can `BLOCK=HASH` be updated to `BLOCK-HASH`

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (sgregglives) [reply docs/Reference/CLI/CLI-Syntax.md]: Removed extra note and changed _option_ to _options_.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 3 above)
Comment (sgregglives) [reply docs/Reference/CLI/CLI-Syntax.md]: The correct syntax format requires the equals sign. I changed the format to better reflect this.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 2 above)
Comment (sgregglives) [reply docs/Reference/CLI/CLI-Syntax.md]: Fixed.

File: docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md [hunk 4]
```diff
@@ -120,5 +120,5 @@ The block parameter can have the following values:
 * `pending` : `tag` - Last block mined plus pending transactions. Use only with [eth_getTransactionCount](../../../Reference/API-Methods.md#eth_gettransactioncount).  
 
 !!! note
-    If [synchronizing in FAST mode](../../../Reference/CLI/CLI-Syntax.md#fast-sync-options), most historical 
-    world state data is unavailable.  Any methods attempting to access unavailable world state data return `null`.
\ No newline at end of file
+    If [synchronizing in FAST mode](../../../Reference/CLI/CLI-Syntax.md#sync-mode), most historical
```
Comment (sgregglives) [inline docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md]: @MadelineMurray As part of removing the sync options section, I had to update this link. However, the link isn't very helpful. I think there's a small information gap that's beyond the scope of this PR. Namely, some kind of explanation of full mode and fast mode could be useful in the docs.

Comment (MadelineMurray) [reply docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md]: This makes sense - could you raise a JI to add this content.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/CLI/CLI-Syntax.md]: I get it now.  I looked at the CLI help - you can specify multiple blocks and hashes.  This needs to be reflected in the doc as well: 

`--required-block, --required-blocks[=BLOCK=HASH[,BLOCK=HASH...]...]`

Comment (sgregglives) [reply docs/Reference/CLI/CLI-Syntax.md]: Updated.

File: docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md (hunk 4 above)
Comment (sgregglives) [reply docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md]: Done.

File: docs/Concepts/NetworkID-And-ChainID.md [hunk 5]
```diff
@@ -24,10 +24,16 @@ same and are specified in the genesis file.
...
+| Network   | Chain ID | Network ID |
+|-----------|----------|------------|
+| `classic` | 61       | 1          |
+| `dev`     | 2018     | 2018       |
+| `goerli`  | 5        | 5          |
+| `kotti`   | 6        | 6          |
+| `mainnet` | 1        | 1          |
+| `mordor ` | 63       | 7          |
+| `rinkeby` | 4        | 4          |
+| `ropsten` | 3        | 3          |
```
Comment (shemnon) [inline docs/Concepts/NetworkID-And-ChainID.md]: I don't think we should alphabetize them.  

Conceptually I feel we should keep the chain families together.  `mainnet`, `ropsten`, `rinkeby`, and `goerli` are all `ETH` chains, `classic`, `kotti`, and `mordor` are all `ETC` chains, then `dev` (an `ETH` chain).  Or maybe dev before the classic chains.  Perhaps a column for 'family' and all are "ETH" except the classic networks.

I think we also need a column for type, mainnet and classic are live or production networks (not sure of the term here).  ropsten, rinkeby, goerli, mordor, and kotti are testnets, dev is a development net.

mainnet
ropsten
rinkeby
goerli
dev
classic
mordor
kotti

File: docs/Reference/API-Methods.md [hunk 6]
```diff
@@ -337,12 +337,17 @@ None
...
+| Chain ID | Network                                        |
+|----------|------------------------------------------------|
+| `61`     | Ethereum Classic network                       |
+| `2018`   | PoW development network                        |
+| `5`      | PoA test network using Clique                  |
+| `6`      | Ethereum Classic PoA test network using Clique |
+| `1`      | Main Ethereum network                          |
+| `63`     | Ethereum Classic PoW test network              |
+| `4`      | PoA test network using Clique                  |
+| `3`      | PoW test network                               |
```
Comment (shemnon) [inline docs/Reference/API-Methods.md]: I think we should sort by chain ID or use the "by family" ordering.  Also a column for the network name would be informative.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 7]
```diff
@@ -659,28 +713,24 @@ network="rinkeby"
...
+| Network   | Description                                                                   |
+|-----------|-------------------------------------------------------------------------------|
+| `classic` | Ethereum Classic network                                                      |
+| `dev`     | PoW development network with a very low difficulty to enable local CPU mining |
+| `goerli`  | PoA test network using Clique                                                 |
+| `kotti`   | Ethereum Classic PoA test network using Clique                                |
+| `mainnet` | Main Ethereum network                                                         |
+| `mordor ` | Ethereum Classic PoW test network                                             |
+| `rinkeby` | PoA test network using Clique                                                 |
+| `ropsten` | PoW test network similar to current main Ethereum network                     |
```
Comment (shemnon) [inline docs/Reference/CLI/CLI-Syntax.md]: Here also, sorted by family, then importance.  It's a subjective sort but I think we should reflect the hierarchy here.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 8]
```diff
@@ -1185,7 +1213,7 @@ Default is 60.
 ### required-block
 
 ```bash tab="Syntax"
---required-block=<BLOCK=HASH>
+--required-block=<BLOCK>=<HASH>
```
Comment (shemnon) [inline docs/Reference/CLI/CLI-Syntax.md]: :+1:

File: docs/Concepts/NetworkID-And-ChainID.md (hunk 5 above)
Comment (sgregglives) [reply docs/Concepts/NetworkID-And-ChainID.md]: I added a Chain column to group together ETH networks and ETC networks and a Type column to label production, test, and development networks.

File: docs/Reference/API-Methods.md (hunk 6 above)
Comment (sgregglives) [reply docs/Reference/API-Methods.md]: I ordered the list by chain ID, and added a Chain column and a Network column for the names.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 7 above)
Comment (sgregglives) [reply docs/Reference/CLI/CLI-Syntax.md]: I added the Chain column and ordered them by importance, according to the list in one of the previous comments, and a Type column.

**Pull request merged**